src/sortStudents.ts

```diff
@@ -1,16 +1,85 @@
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade,
 }
 
 // create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+function getAverageGrade(grades: number[]): number {
+  return grades.reduce((prev, current) => prev + current, 0) / grades.length;
+}
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+) : Student[] {
+  const copyStudents: Student[] = [...students];
+
+  switch (sortBy) {
+    case SortType.Name:
+      if (order === 'asc') {
+        copyStudents.sort((a: Student, b: Student) => (a.name)
+          .localeCompare(b.name));
+      } else {
+        copyStudents.sort((a: Student, b: Student) => (b.name)
+          .localeCompare(a.name));
+      }
+      break;
+
+    case SortType.Surname:
+      if (order === 'asc') {
+        copyStudents.sort((a: Student, b: Student) => (a.surname)
+          .localeCompare(b.surname));
+      } else {
+        copyStudents.sort((a: Student, b: Student) => (b.surname)
+          .localeCompare(a.surname));
+      }
```

Comment:
```suggestion
  switch (sortBy) {
    case SortType.Name:
    case SortType.Surname:
      order === 'asc'
      ? your code if true
      : your code if false
      
      break;

```

You can combine cases with the same logic

Reply:
Thank you so much!)